src/itype.cpp

```diff
@@ -4,14 +4,25 @@
 #include "game.h"
 #include "item_factory.h"
 #include "translations.h"
+#include "output.h"
 
 #include <fstream>
 #include <stdexcept>
 #include <algorithm>
 
 std::string itype::nname( unsigned int const quantity ) const
 {
-    return ngettext( name.c_str(), name_plural.c_str(), quantity );
+    std::string str( ngettext( name.c_str(), name_plural.c_str(), quantity ) );
+
+    if( wheel ) {
+        return string_format( str, wheel->diameter );
+
+    } else if( engine && engine->displacement > 0 ) {
+        return string_format( str, engine->displacement / 100.0 );
```

Comment:
Rather dangerous. `string_format()` expects exact number of arguments. Why do you need to format a string using json values from the very same source? The whole idea looks like an unnecessary coupling.